core/src/commonMain/kotlin/com/ustadmobile/core/controller/ContentEntryEdit2Presenter.kt

```diff
@@ -0,0 +1,165 @@
+                                 systemImpl: UstadMobileSystemImpl,
+                                 db: UmAppDatabase, repo: UmAppDatabase,
+                                 private val containerDownloadManager: ContainerDownloadManager?,
+                                 activeAccount: DoorLiveData<UmAccount?> = UmAccountManager.activeAccountLiveData)
+    : UstadEditPresenter<ContentEntryEdit2View, ContentEntryWithLanguage>(context, arguments, view, lifecycleOwner, systemImpl,
+        db, repo, activeAccount) {
+
+    enum class LicenceOptions(val optionVal: Int, val messageId: Int){
+        LICENSE_TYPE_CC_BY(ContentEntry.LICENSE_TYPE_CC_BY, MessageID.licence_type_cc_by),
+        LICENSE_TYPE_CC_BY_SA(ContentEntry.LICENSE_TYPE_CC_BY_SA, MessageID.licence_type_cc_by_sa),
+        LICENSE_TYPE_CC_BY_SA_NC(ContentEntry.LICENSE_TYPE_CC_BY_SA_NC, MessageID.licence_type_cc_by_sa_nc),
+        LICENSE_TYPE_CC_BY_NC(ContentEntry.LICENSE_TYPE_CC_BY_NC, MessageID.licence_type_cc_by_nc),
+        ALL_RIGHTS_RESERVED(ContentEntry.ALL_RIGHTS_RESERVED, MessageID.licence_type_all_rights),
+        LICENSE_TYPE_CC_BY_NC_SA(ContentEntry.LICENSE_TYPE_CC_BY_NC_SA, MessageID.licence_type_cc_by_nc_sa),
+        LICENSE_TYPE_PUBLIC_DOMAIN(ContentEntry.LICENSE_TYPE_PUBLIC_DOMAIN, MessageID.licence_type_public_domain),
+        LICENSE_TYPE_OTHER(ContentEntry.LICENSE_TYPE_OTHER, MessageID.licence_type_other)
+    }
+
+    data class UmStorageOptions(var messageId: Int,var label: String)
+
+    private var parentEntryUid:Long = 0
+
+
+    open class StorageOptions(context: Any, val storage: UmStorageOptions): MessageIdOption(storage.messageId,context){
+        override fun toString(): String {
+            return storage.label
+        }
+    }
+
+    class LicenceMessageIdOptions(licence: LicenceOptions,context: Any)
+        : MessageIdOption(licence.messageId,context, licence.optionVal)
+
+    override val persistenceMode: PersistenceMode
+        get() = PersistenceMode.DB
+
+    private var storageOptions: List<UMStorageDir>? = null
+
+    override fun onCreate(savedState: Map<String, String>?) {
+        super.onCreate(savedState)
+        view.licenceOptions = LicenceOptions.values().map { LicenceMessageIdOptions(it, context) }
+        parentEntryUid = arguments[ARG_PARENT_ENTRY_UID]?.toLong()?:0
+        systemImpl.getStorageDirs(context, object : UmResultCallback<List<UMStorageDir>> {
+            override fun onDone(result: List<UMStorageDir>?) {
+                storageOptions = result
+                if(result != null){
+                   view.runOnUiThread(Runnable {
+                       view.setUpStorageOptions(result)
+                   })
+                }
+            }
+        })
+    }
+
+    override suspend fun onLoadEntityFromDb(db: UmAppDatabase): ContentEntryWithLanguage? {
+        val entityUid = arguments[ARG_ENTITY_UID]?.toLong() ?: 0
+        return withTimeoutOrNull(2000) {
```

Comment:
As per comment on the Fragment for this, let's change this to set the leaf property according to the argument